runtime/compiler/optimizer/HotFieldMarking.cpp

```diff
@@ -162,7 +156,7 @@ int32_t TR_HotFieldMarking::perform()
                {
                int32_t classNameLength = 0;
                char *className = comp()->fej9()->getClassNameChars(itr->second->_clazz, classNameLength);
-               traceMsg(comp(),"Hot field marked or updated. signature: %.*s; class:%.*s; fieldName: %.*s; frequencyScore = %d; fieldOffset: %d; \n", J9UTF8_LENGTH(itr->second->_fieldSig), J9UTF8_DATA(itr->second->_fieldSig), J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(itr->second->_fieldName), J9UTF8_DATA(itr->second->_fieldName), itr->second->_score, fieldOffset);
+               printf("Hot field marked or updated - signature:%.*s;   class:%.*s;   fieldName:%.*s;   frequencyScore=%d;   fieldOffset:%d \n", J9UTF8_LENGTH(itr->second->_fieldSig), J9UTF8_DATA(itr->second->_fieldSig), classNameLength, J9UTF8_DATA(className), J9UTF8_LENGTH(itr->second->_fieldName), J9UTF8_DATA(itr->second->_fieldName), itr->second->_score, fieldOffset);
```

Comment:
I'm not clear of the reason why this needs to go to stdout now rather than into the log file (even if it is guarded with a trace option).  This generally isn't the way tracing/logging is done in the JIT.  Can you provide more clarity why the log file is insufficient?

Reply:
It was done to remain consistent with other usages of `TR_TraceMarkingOfHotFields` found in the JIT that already exist and use `stdout` for output. Is this outdated?

Reply:
It was likely never done correctly in the first place then.  I'd say it is an anti-pattern.  If tracing output goes straight to stdout we lose our ability to control and capture it.

Reply:
Ok I see, looks like I had it right originally. I will go back to `traceMsg` and stay consistent with the current practice now.